src/main/java/edu/wright/airviewer2/AIRViewerController.java

```diff
@@ -392,6 +407,46 @@ public void handle(ActionEvent e) {
 		refreshUserInterface();
 		return model;
 	}
+	
+	/**
+     * This method extracts an integer value from the navigate input control
+     * If the input is not an integer then the method catches a TypeError exception
+     * If the input is indeed an integer then the method sets the new page index
+     * provided it's a valid page index
+     * All errors are sent as output to navigationWarning control
+     */
+    private void navigateToPage() {
+        try {
+            int page = Integer.parseInt(navigateInput.getText());
+
+            if (page <= 0) {
+                navigateWarning.setText("Invalid input : less than 1");
+                return;
+            }
+
+            if (page > pagination.getPageCount()) {
+                navigateWarning.setText("Invalid input : select under " + (pagination.getPageCount() + 1));
+                return;
+            }
+
+            pagination.setCurrentPageIndex(page - 1);
+
+            navigateWarning.setText("");
+        } catch (Exception e) {
+            System.out.println(e);
+            MessageBox.show(e.toString(), "Exception");
```

Comment:
instead of adding the raw exception message add user-friendly messages to the UI in message box